packages/adp-tooling/src/prompt/changes/prompts/change-data-source.ts

```diff
@@ -0,0 +1,298 @@
+    }
+    const annotationUri = data.oDataAnnotations[selectOdataSourceAnnotation];
+    return shouldShowODataVersionInfoMessage({
+        currentURI: annotationUri,
+        isForV2: data.isForV2,
+        replaceURI: data.replaceURI,
+        isCFEnv: data.isCFEnv
+    });
+}
+
+/**
+ * Gets the questions for changing the data source.
+ *
+ * @param {AdpChangeDataSourceQuestions} data - the data for the questions
+ * @returns {Question<AdpChangeDataSourceAnswers>[]} - the questions
+ */
+export function getQuestions(data: AdpChangeDataSourceQuestions): Question<AdpChangeDataSourceAnswers>[] {
+    return [
+        {
+            type: 'input',
+            name: 'isInSafeMode',
+            message: t('prompts.isInSafeModeLabel'),
+            guiOptions: {
+                type: 'label'
+            },
+            when: () => data.isInSafeMode && data.isYUI
+        } as InputQuestion<AdpChangeDataSourceAnswers>,
+        {
+            type: 'list',
+            name: 'oDataSource',
+            message: t('prompts.oDataSourceLabel'),
+            choices: data.oDataSources?.map((dS) => dS.dataSourceName) ?? [],
+            default: data.oDataSources?.[0]?.dataSourceName,
+            store: false,
+            guiOptions: {
+                mandatory: true,
+                hint: t('prompts.oDataSourceTooltip')
+            },
+            validate: (value: string) => validateODataServices(value, data.oDataSources)
+        } as ListQuestion<AdpChangeDataSourceAnswers>,
+        {
+            type: 'input',
+            name: 'oDataSourceURI',
+            message: t('prompts.oDataSourceURILabel'),
+            guiOptions: {
+                mandatory: true,
+                hint: t('prompts.oDataSourceURITooltip')
+            },
+            validate: (value: string) => validateURI(value, t('prompts.oDataSourceURILabel')),
+            when: !!data.oDataSources.length,
+            store: false
+        } as InputQuestion<AdpChangeDataSourceAnswers>,
+        {
+            type: 'input',
+            name: 'oDataURIInfo',
+            message: () => {
+                let v2Prefix: string, v4Prefix: string;
+                if (data.isCFEnv) {
+                    v2Prefix = ODATA_V2_PREFIX_CF;
+                    v4Prefix = ODATA_V4_PREFIX_CF;
+                } else {
+                    v2Prefix = ODATA_V2_PREFIX_ABAP;
+                    v4Prefix = ODATA_V4_PREFIX_ABAP;
+                }
+                return t('prompts.oDataURIInfoLabel', { v2Prefix, v4Prefix });
+            },
+            when: (answers: Answers) => data.isYUI && shouldShowODataUriInfo(answers.oDataSourceURI, data.isCFEnv),
+            guiOptions: {
+                type: 'label',
+                applyDefaultWhenDirty: true
+            }
+        } as InputQuestion<AdpChangeDataSourceAnswers>,
+        {
+            type: 'input',
+            name: 'oDataURIV2Info',
+            message: () => {
+                const versionOne = 'V2';
```

Comment:
We have a constant/type for odata version (I think in project-access) please reuse it.

Reply:
Fixed with commit : fcb2eb3c603db4f005757221894faf5bf36e714e